### PR TITLE
applications: nrf_desktop: DFU through the external flash

### PR DESCRIPTION
**Pull request by pdunaj**
Jira:DESK-922

### CODE REVIEW

File: applications/nrf_desktop/configuration/nrf52840dk_nrf52840/pm_static_ZDebugMCUBootQSPI.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  address: 0xc000
+  size: 0xd4000
+  region: flash_primary
+mcuboot_primary_app:
+  orig_span: &id002
+  - app
+  span: *id002
+  address: 0xc200
+  size: 0xd3e00
+mcuboot_scratch:
```
Comment (MarekPieta) [inline applications/nrf_desktop/configuration/nrf52840dk_nrf52840/pm_static_ZDebugMCUBootQSPI.yml:25]: MCUBoot no longer uses scratch partition

File: applications/nrf_desktop/configuration/nrf52840dk_nrf52840/app_ZDebugMCUBootQSPI.conf [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+CONFIG_BOOTLOADER_MCUBOOT=y
+
+CONFIG_IMG_MANAGER=y
+CONFIG_IMG_BLOCK_BUF_SIZE=0
+CONFIG_MCUBOOT_IMG_MANAGER=y
+
+CONFIG_NORDIC_QSPI_NOR=y
+CONFIG_NORDIC_QSPI_NOR_FLASH_LAYOUT_PAGE_SIZE=4096
+CONFIG_NORDIC_QSPI_NOR_FLASH_ALLOW_STACK_USAGE_FOR_DATA_IN_FLASH=y
```
Comment (MarekPieta) [inline applications/nrf_desktop/configuration/nrf52840dk_nrf52840/app_ZDebugMCUBootQSPI.conf:203]: Should this be part of the application configuration? Bootloader performs copying data between FLASH areas.

File: applications/nrf_desktop/configuration/nrf52840dk_nrf52840/pm_static_ZDebugMCUBootQSPI.yml (hunk 1 above)
Comment (pdunaj) [reply applications/nrf_desktop/configuration/nrf52840dk_nrf52840/pm_static_ZDebugMCUBootQSPI.yml:25]: I will get rid of it

File: applications/nrf_desktop/configuration/nrf52840dk_nrf52840/app_ZDebugMCUBootQSPI.conf (hunk 2 above)
Comment (pdunaj) [reply applications/nrf_desktop/configuration/nrf52840dk_nrf52840/app_ZDebugMCUBootQSPI.conf]: moved

File: applications/nrf_desktop/configuration/nrf52840dk_nrf52840/pm_static_ZDebugMCUBootQSPI.yml (hunk 1 above)
Comment (pdunaj) [reply applications/nrf_desktop/configuration/nrf52840dk_nrf52840/pm_static_ZDebugMCUBootQSPI.yml]: removed - sizes updated

**Pull request merged**